src/coreclr/minipal/Windows/dn-u16.cpp

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <Windows.h>
+#include <wchar.h>
+
+#include <dn-u16.h>
+
+size_t strlen_u16(const WCHAR* str)
+{
+    if (str == NULL)
+        return 0;
+    return ::wcslen(str);
+}
+
+int strcmp_u16(const WCHAR* str1, const WCHAR* str2)
+{
+    return ::wcscmp(str1, str2);
+}
+
+int strncmp_u16(const WCHAR* str1, const WCHAR* str2, size_t count)
+{
+    return ::wcsncmp(str1, str2, count);
+}
+
+WCHAR* strcpy_u16(WCHAR* dst, size_t dstLen, const WCHAR* src)
```

Comment:
Should we keep the `_s` suffix?

Reply:
I would prefer not to. We own the abstraction here and the `_s` suffix is just weird. Our internal APIs should all be "safe"/"secure".

Reply:
The less secure versions without `_s` suffix are still fine to use for some of the APIs.

I think we should have a simple decoder ring for these APIs: make them exactly same as the `char` version, with `u16` suffix and `char` changed to `char16_t`. If we start improving names or behaviors of these APIs, it will become muddy.

It may look better if the u16 prefix is first, like `u16_strcpy_s`. People out there tend to prefer `u16_` prefix: https://github.com/search?q=strlen_u16&type=code has 40 hits, https://github.com/search?q=u16_strlen&type=code has 924 hits.

Reply:
> The less secure versions without _s suffix are still fine to use for some of the APIs.

Agreed, but the point here is an abstraction that we want to use internally. I still have to go through the codebase and update all the `_s` APIs too. I'd prefer to simply state the API name rather than perpetuate the historical artifact of where they came from.

> It may look better if the u16 prefix is first, like u16_strcpy_s. People out there tend to prefer u16_ prefix:

That is fair and I will apply that.

I still don't like the `_s` there. It makes it more convenient for someone to write the less secure one when they just shouldn't be used. I don't want a full projection of standard APIs just because someone doesn't want to deal with the secure API. I don't see the decoder ring issue at all. The APIs are "secure" by definition rather than mimicing the standard history.